uber/models.py

```diff
@@ -160,6 +162,110 @@ def process_bind_param(self, value, dialect):
 default_metadata = MetaData(naming_convention=immutabledict(default_naming_convention))
 
 
+def JSONColumnMixin(column_name, fields, admin_only=False):
+    """
+    Creates a new mixin class with a JSON column named column_name.
+
+    The newly created mixin class will have a SQLAlchemy JSON Column, named
+    `column_name`, along with two other attributes column_name_fields and
+    column_name_qualified_fields which describe the fields that the JSON
+    Column is expected to hold.
+
+    For example::
+
+        >>> SocialMediaMixin = JSONColumnMixin('social_media', ['Twitter', 'LinkedIn'])
+        >>> SocialMediaMixin.social_media
+        Column('social_media', JSON(), table=None, nullable=False, default=ColumnDefault({}), server_default=DefaultClause('{}', for_update=False))
+        >>> SocialMediaMixin.social_media_fields
+        OrderedDict([('twitter', 'Twitter'), ('linked_in', 'LinkedIn')])
+        >>> SocialMediaMixin.social_media_qualified_fields
+        OrderedDict([('social_media__twitter', 'twitter'), ('social_media__linked_in', 'linked_in')])
+
+    Instances of the newly created mixin class have convenience accessors for
+    the attributes defined by `fields`, both directly and using their fully
+    qualified forms::
+
+        >>> sm = SocialMediaMixin()
+        >>> sm.twitter = 'https://twitter.com/MAGFest'  # Get and set "twitter" directly
+        >>> sm.twitter
+        'https://twitter.com/MAGFest'
+        >>> sm.social_media__twitter  # Get and set qualified "social_media__twitter"
+        'https://twitter.com/MAGFest'
+        >>> sm.social_media__twitter = '@MAGFest'
```

Comment:
It's entirely possible that there could be a naming collision between the JSON fields and an existing field on a class. Like, if the `PanelistApplicant` *already* had a column named `twitter`, then the `twitter` property for the `SocialMediaMixin` would be inaccessible. The qualified versions of the JSON fields ensure that there will always be a collision free way to access those fields.